Source/WebKit/NetworkProcess/NetworkResourceLoadParameters.cpp

```diff
@@ -136,7 +136,7 @@ RefPtr<SecurityOrigin> NetworkResourceLoadParameters::parentOrigin() const
 {
     if (frameAncestorOrigins.isEmpty())
         return nullptr;
-    return frameAncestorOrigins.first();
+    return frameAncestorOrigins.first().ptr();
```

Comment:
After this change, `parentOrigin()` will no longer return nullptr, which changes the behavior of this function, and existing call sites need to be updated accordingly (e.g. if they do null check it will not work).

According to the commit message, the crash is caused by converting null origin to string, can we just skip nullptr instead? It will be less risky

Reply:
`frameAncestorOrigins` could still be empty right? So this function can still return nullptr.

I don't think there is a behavior change here when site isolation is disabled. The only time we would append nullptr to this vector was for remote frames.

Reply:
Yeah, this doesn't change behavior with site isolation off, and it's at least a step in the right direction with site isolation on.  There may be a place or two where we do the wrong thing for opaque origins which would only affect site isolation on, but if we come across them we'll fix them.

Reply:
`parentOrigin` used to return nullptr when the first element is nullptr but now it does not, so technically call sites that check nullptr should also check for first opaque origin instead.
If we are sure this only affects site isolation, then it's probably fine now, and watch out for potential breakage.

Reply:
The first element could only be nullptr if there was a RemoteFrame, which only happens with site isolation on.